hopp/offshore/floating_platform.py

```diff
@@ -0,0 +1,357 @@
+        
+        # Add in the mass of the substructure to total mass (may or may not impact the final install cost)
+        _, substructure_mass = calc_substructure_mass_and_cost(self.mass, self.area, 
+                        self.depth, fab_cost_rate, design_cost, steel_cost
+                        )
+
+        total_mass = self.mass + substructure_mass  # t
+
+         # Call the install_platform function
+        self.install_capex = install_platform(total_mass, self.area, self.distance, \
+                                                   install_duration, self.install_vessel)
+
+    # An install object needs to have attribute system_capex, installation_capex, and detailed output
+    @property
+    def system_capex(self):
+
+        return {}
+
+    @property 
+    def installation_capex(self):
+        
+        return self.install_capex
+
+    @property
+    def detailed_output(self):
+
+        return {}
+
+# Define individual calculations and functions to use outside or with ORBIT
+def calc_substructure_mass_and_cost(mass, area, depth, fab_cost_rate=14500., design_cost=4.5e6, sub_cost_rate=3000):
+    '''
+    calc_substructure_mass_and_cost returns the total mass including substructure, topside and equipment.  Also returns the cost of the substructure and topside
+    Inputs: mass            | Mass of equipment on platform (tonnes)
+            area            | Area needed for equipment (meter^2) (not necessary)
+            depth           | Ocean depth at platform location (meters)
+            fab_cost_rate   | Cost rate to fabricate topside (USD/tonne)
+            design_cost     | Design cost to design structural components (USD) from ORBIT
+            sub_cost_rate   | Steel cost rate (USD/tonne) from ORBIT'''
+
+    '''
+    Platform is substructure and topside combined
+    All functions are based off NREL's ORBIT (oss_design)
+    default values are specified in ORBIT
+    '''
+    topside_mass = mass
+    topside_fab_cost_rate   =   fab_cost_rate   
+    topside_design_cost     =   design_cost
+
+    '''Topside Cost & Mass
+    Topside Mass is the required Mass the platform will hold
+    Topside Cost is a function of topside mass, fab cost and design cost'''
+    topside_cost   =   topside_mass*topside_fab_cost_rate + topside_design_cost
+
+    '''Substructure
+    Substructure Mass is a function of the topside mass
+    Substructure Cost is a function of of substructure mass pile mass and cost rates for each'''
+
+    substructure_cost_rate  =   sub_cost_rate        # USD/t
+
+    substructure_mass       =   0.4*topside_mass        # t
+    substructure_cost       =   (substructure_mass*substructure_cost_rate)     # USD  
+    substructure_total_mass =   substructure_mass       # t
+
+    '''These arrays were pulled from Rebecca Fuchs SemitTaut_mooring cost models'''
+    depths = np.array([0, 500, 750, 1000, 1250, 1500]) # [m]
+
+    anchor_costs = np.array([28823, 112766, 125511, 148703, 204988, 246655]) # [USD]
```

Comment:
Having these copied over and hard coded makes me nervous. They don't appear to have a year associated with them or a source and will certainly change with time. Does ORBIT give a source for these? Is it possible to use ORBIT more directly so we don't have so much copy-paste action going on and can stay update more easily?